core/types/calls.cc

```diff
@@ -1567,6 +1567,23 @@ class Magic_callWithSplatAndBlock : public IntrinsicMethod {
     }
 } Magic_callWithSplatAndBlock;
 
+class Magic_suggestUntypedConstantType : public IntrinsicMethod {
+public:
+    void apply(Context ctx, DispatchArgs args, const Type *thisType, DispatchResult &res) const override {
+        ENFORCE(args.args.size() == 1);
+        auto ty = args.args.front()->type;
+        auto loc = args.locs.args[0];
+        if (!ty->isUntyped()) {
+            if (auto e = ctx.state.beginError(loc, core::errors::Infer::UntypedConstantSuggestion)) {
+                e.setHeader("Suggested type: `{}`", ty->show(ctx));
```

Comment:
Might be nice to connect this to the constant missing type error that will have already been reported above.

Reply:
Ack, I don't see how to do this easily.
I might remove the error on the other end completely though...